src/azure_cosmosdb/cosmosdb_utils.bal

```diff
@@ -0,0 +1,190 @@
+            url = url + path;
+        }
+    }
+    return <@untainted> url;
+}
+
+# To create a custom error instance
+# + return - returns error.  
+isolated function prepareError(string message, error? err = ()) returns error { 
+    error azureError;
+    if (err is error) {
+        azureError = AzureError(message, err);
+    } else {
+        azureError = AzureError(message);
+    }
+    return azureError;
+}
+
+# To attach required basic headers to call REST endpoint
+# + req - http:Request to add headers to
+# + host - 
+# + keyToken - master or resource token
+# + tokenType - denotes the type of token: master or resource.
+# + tokenVersion - denotes the version of the token, currently 1.0.
+# + params - an object of type HeaderParamaters
+# + return - If successful, returns same http:Request with newly appended headers. Else returns error.  
+isolated function setHeaders(http:Request req, string host, string keyToken, string tokenType, string tokenVersion,
+RequestHeaderParameters params) returns http:Request|error {
+    req.setHeader("x-ms-version",params.apiVersion);
+    req.setHeader("Host",host);
+    req.setHeader("Accept","*/*");
+    req.setHeader("Connection","keep-alive");
+
+    string?|error date = getTime();
+    if date is string
+    {
+        string? s = generateTokenNew(params.verb,params.resourceType,params.resourceId,keyToken,tokenType,tokenVersion);
+        req.setHeader("x-ms-date",date);
+        if s is string {
+            req.setHeader("Authorization",s);
+        } else {
+            return prepareError("Authorization token is null");
+        }
+    } else {
+        return prepareError("Date header is invalid/null");
+    }
+    return req;
+}
+
+# To construct the hashed token signature for a token 
+# + return - If successful, returns string representing UTC date and time 
+#               (in "HTTP-date" format as defined by RFC 7231 Date/Time Formats). Else returns error.  
+isolated function getTime() returns string?|error {
+    time:Time time1 = time:currentTime();
+    var time2 = check time:toTimeZone(time1, "Europe/London");
+    string|error timeString = time:format(time2, "EEE, dd MMM yyyy HH:mm:ss z");
+    return timeString;
+}
+
+# To construct the hashed token signature for a token to set  'Authorization' header
+# + verb - HTTP verb, such as GET, POST, or PUT
+# + resourceType - identifies the type of resource that the request is for, Eg. "dbs", "colls", "docs"
+# + resourceId -dentity property of the resource that the request is directed at
+# + keyToken - master or resource token
+# + tokenType - denotes the type of token: master or resource.
+# + tokenVersion - denotes the version of the token, currently 1.0.
+# + return - If successful, returns string which is the  hashed token signature. Else returns ().  
+isolated function generateTokenNew(string verb, string resourceType, string resourceId, string keyToken, string tokenType, 
+string tokenVersion) returns string? {
```

Comment:
Shall we format it properly?
(Can move the tokenVersion also to the previous line if id doesn't exceed 120 characters)